src/MahApps.Metro/MahApps.Metro.Shared/Controls/Helper/TextBoxHelper.cs

```diff
@@ -301,23 +336,48 @@ private static void TextBoxBaseContextMenuOpening(object sender, ContextMenuEven
                 var separatorMenuItem2 = new Separator();
                 tbBase.ContextMenu.Items.Insert(cmdIndex, separatorMenuItem2);
             }
+            AddExtraItemsToContextMenu(tbBase, true);
+            tbBase.ContextMenu.Visibility = tbBase.ContextMenu.Items.Count > 0 ? Visibility.Visible : Visibility.Hidden;
+        }
+
+        private static void AddExtraItemsToContextMenu(TextBoxBase tbBase, bool shouldAddSeparator = false)
+        {
+            List<MenuItem> extraMenuItems = (List<MenuItem>)tbBase.GetValue(ExtraContextMenuItemsProperty);
+            if (extraMenuItems != null && extraMenuItems.Count > 0)
+            {
+                // Add separator if necessary, then extra items to menu
+                if (shouldAddSeparator)
```

Comment:
I think we should not add a default separator, cause users can't avoid this. If the user want it then he should do this itself.